Plugins/Public/minecontrol_plugin/Main.cpp

```diff
@@ -568,183 +428,283 @@ void __stdcall SPMunitionCollision(struct SSPMunitionCollisionInfo const & ci, u
 	uint iClientSystemID;
 	pub::Player::GetSystem(iClientID, iClientSystemID);
 	CmnAsteroid::CAsteroidSystem* csys = CmnAsteroid::Find(iClientSystemID);
-	if (csys)
+	if (!csys)
+		return;
+
+	// Find asteroid field that matches the best.
+	for (CmnAsteroid::CAsteroidField* cfield = csys->FindFirst(); cfield; cfield = csys->FindNext())
 	{
-		// Find asteroid field that matches the best.
-		for (CmnAsteroid::CAsteroidField* cfield = csys->FindFirst(); cfield; cfield = csys->FindNext())
+		if(!cfield->near_field(vPos))
+			continue;
+		const Universe::IZone *zone = cfield->get_lootable_zone(vPos);
+		if (!zone || !zone->lootableZone)
+			continue;
+
+		const auto& zoneBonusData = set_mapZoneBonus.find(zone->iZoneID);
+		if(zoneBonusData != set_mapZoneBonus.end())
 		{
-			try
+			auto& zoneData = zoneBonusData->second;
+			if (zoneData.fCurrReserve == 0.0f)
+				return;
+
+			if (zoneData.iReplacementLootID)
+				lootId = zoneData.iReplacementLootID;
+
+			itemCount *= zoneData.fMultiplier;
+
+			itemCount = max(itemCount, zoneData.fCurrReserve);
+			zoneData.fCurrReserve -= itemCount;
+			zoneData.fMined += itemCount;
+		}
+
+		itemCount *= GetBonus(cd.equippedID, lootId) * set_globalModifier;
+		itemCount += cd.overminedFraction; // add the decimal remainder from last mining eevnt.
```

Comment:
Add by fraction? Is that intended

Reply:
Basically we're adding a leftover from the last mining event, basically the way it works is:
Mined amount with modifiers sums up to 7.4, we add 7 to the player's cargo hold, and store 0.4 in the 'overminedFraction'
We then add it to the mined amount in the next event, so that if next event would yield 8.6, with the fraction it'd go up to 9.0 and give the player 9 units.

It's to make small percentage boosts in yield be properly reflected despite integer-based nature of the player's cargo.